### PR TITLE
refactor

### PR DESCRIPTION
**Pull request by tonicbupt**
* use `urfave.cli` instead of `flag`
* pass `Config` to `Sync` and further to `object.ObjectStorage` so some flags can be passed too
* use `copyObject` instead of `copy` which is a built-in function name
* add version information in cli

### CODE REVIEW

**Comment by tonicbupt** [comment]
actually i think we can abandon `vendor` directory now ... everyone should use `go mod`

**Comment by yujunz** [comment]
> actually i think we can abandon `vendor` directory now ... everyone should use `go mod`

Didn't we tried it once? I roughly remember it was blocked by some specific module. I'm OK to use `go mod` as long as it works.

File: VERSION [hunk 1]
```diff
@@ -0,0 +1 @@
+v0.0.5
```
Comment (yujunz) [inline VERSION]: Is this compatible with goreleaser which depends on git tag for versioning ?

**Comment by davies** [comment]
@tonicbupt Can we split this into two PR (one for removed vendor/ and another one for cli options)?

**Comment by yujunz** [comment]
Maybe three...

* vendor/mod
* cli options
* versioning

**Comment by tonicbupt** [comment]
@davies @yujunz reverted vendor commit, it was actually the output of `go mod vendor`. so currently it'll be splitted into 2 PRs, vendor and cli+version

File: VERSION (hunk 1 above)
Comment (tonicbupt) [reply VERSION]: ha? I just took the latest release version here...

Comment (yujunz) [reply VERSION]: Please refer to https://goreleaser.com/environment/ for getting version. We are using it to make releases in GitHub Actions.

> Default wise GoReleaser sets three ldflags:
> 
> * main.version: Current Git tag (the v prefix is stripped) or the name of the snapshot, if you're using the --snapshot flag
> * main.commit: Current git commit SHA
> * main.date: Date according RFC3339
> 
> You can use it in your main.go file:
> 
> ```
> package main
> 
> import "fmt"
> 
> var (
> 	version = "dev"
> 	commit  = "none"
> 	date    = "unknown"
> )
> 
> func main() {
>   fmt.Printf("%v, commit %v, built at %v", version, commit, date)
> }
> ```
> 
> You can override this by changing the ldflags option in the build section.
>

Comment (tonicbupt) [reply VERSION]: checked goreleaser docs and got it, i'll do some changes in next PR

File: .goreleaser.yml [hunk 2]
```diff
@@ -1,13 +1,19 @@
 builds:
-- env:
-  - CGO_ENABLED=0
+  - env:
+      - CGO_ENABLED=0
+    ldflags:
+      - -s -w 
+      - -X github.com/juicedata/juicesync/versioninfo.VERSION={{.Version}}
+      - -X github.com/juicedata/juicesync/versioninfo.REVISION={{.Commit}}
+      - -X github.com/juicedata/juicesync/versioninfo.REVISIONDATE={{.Date}}
```
Comment (tonicbupt) [inline .goreleaser.yml]: @yujunz I changed `ldflags` here to get adapted to goreleaser

Comment (tonicbupt) [reply .goreleaser.yml]: and also removed that `VERSION` file which turned out to be unnecessary

**Pull request merged**